ran-core/src/main/java/io/ran/Clazz.java

```diff
@@ -466,15 +464,28 @@ public ClazzMethodList methods() {
 					result.put(m, new ClazzMethod(this, m));
 				});
 			} else {
+				Clazz finalWorking = working;
 				Arrays.stream(working.clazz.getDeclaredMethods()).filter(m -> !m.isBridge()).forEach(m -> {
-					result.put(m, new ClazzMethod(this, m));
+					result.put(m, new ClazzMethod(this, finalWorking, m));
+				});
+				Arrays.stream(working.clazz.getInterfaces())
+						.flatMap(inter -> Stream.of(inter.getMethods()))
+						.filter(m -> !m.isBridge())
+						.filter(m -> result.keySet().stream().noneMatch(otherM -> getSignatureOfMethod(m).equals(getSignatureOfMethod(otherM))))
+						.forEach(m -> {
+					result.put(m, new ClazzMethod(this, finalWorking, m));
 				});
 			}
 			working = working.getSuper();
 		} while (working.clazz != null && !Object.class.equals(working.clazz));
 		return new ClazzMethodList(result.values());
 	}
 
+	private static String getSignatureOfMethod(Method method) {
+		// TODO: What about static/final/etc.? method.getModifiers()
+		return method.getReturnType().getName()+" "+method.getName()+"("+Stream.of(method.getParameters()).map(p->p.getType().getName()).collect(Collectors.joining(","));
+	}
+
```

Comment:
There is no test to validate that a static method and none-static method which otherwise have the same signature cannot co-exist (maybe the JVM doesn't allow it... so not really a problem).